src/Microsoft.PowerShell.Commands.Utility/commands/utility/TestJsonCommand.cs

```diff
@@ -0,0 +1,94 @@
+using System;
+using System.Collections.Generic;
+using System.Management.Automation;
+using System.Management.Automation.Internal;
+
+using Newtonsoft.Json;
+using Newtonsoft.Json.Linq;
+using Newtonsoft.Json.Schema;
+
+namespace Microsoft.PowerShell.Commands
+{
+    /// <summary>
+    /// This class implements Test-Json command.
+    /// </summary>
+    [Cmdlet(VerbsDiagnostic.Test, "Json", HelpUri = "")]
+    public class TestJsonCommand : PSCmdlet
+    {
+        /// <summary>
+        /// An Json to be validated.
+        /// </summary>
+        [Parameter(Position = 0, Mandatory = true, ValueFromPipeline = true)]
+        public String Json { get; set; }
+
+        /// <summary>
+        /// A schema to validate the Json against.
+        /// </summary>
+        [Parameter(Position = 1)]
+        [ValidateNotNullOrEmpty()]
+        public String Schema { get; set; }
+
+        private JSchema _jschema = null;
+
+        /// <summary>
+        /// Prepare an Json schema.
+        /// </summary>
+        protected override void BeginProcessing()
+        {
+            if (Schema != null)
```

Comment:
I don't think this check is necessary since Schema property is validated as not null.

Reply:
It is a check that the property present.

Reply:
This is a required property.  Json check cannot be run without a valid schema property.  Make this property "mandatory" and let PowerShell do the validation rather than have an explicit check.

Reply:
What if an user want only check json parsing? Why we should block the scenario?

Reply:
Ok, I understand now.  Can you add a comment making it clear that Schema is optional and without it Json parsing is still performed and of value?  I assume this validates that the json is well formed.

Reply:
I added the comment in Schema parameter description and in the PR description.